tools/src/wyvern/target/corewyvernIL/expression/FieldSet.java

```diff
@@ -48,11 +54,27 @@ public void setExprToAssign(Expression exprToAssign) {
 	}
 
 	@Override
-	public ValueType typeCheck(TypeContext env) {
-		// TODO Auto-generated method stub
-		return null;
+	public ValueType typeCheck(TypeContext ctx) {
+		
+		// Figure out types of object and expression.
+		ValueType valTypeExpr = exprToAssign.typeCheck(ctx);
+		StructuralType structTypeObj = objectExpr.typeCheck(ctx).getStructuralType(ctx);
+		
+		// Figure out the type of the field.
+		DeclType declTypeField = structTypeObj.findDecl(fieldName, ctx);
+		if (!(declTypeField instanceof VarDeclType))
+			throw new RuntimeException("Type error: trying to set the field " + fieldName + ", but this isn't"
+										+ "a var declaration.");
```

Comment:
I know a lot of the current code throws RuntimeExceptions on type errors, but I'd like to move toward new code reporting errors in a better way.  Would you mind replacing this with code like what's in UnresolvedType.getILType()?  i.e. use ToolError.reportError, using an appropriate message from the ErrorMessage enum.